s3/src/test/io/ddf/s3/S3DDFManagerTests.java

```diff
@@ -0,0 +1,83 @@
+package io.ddf.s3;
+
+import io.ddf.DDFManager;
+import io.ddf.datasource.DataFormat;
+import io.ddf.datasource.S3DataSourceCredentials;
+import io.ddf.datasource.S3DataSourceDescriptor;
+import io.ddf.datasource.S3DataSourceURI;
+import io.ddf.exception.DDFException;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+
+public class S3DDFManagerTests {
+
+    public static S3DDFManager manager;
+    public static Logger LOG;
+
+    @Test
+    public void testDDFConfig() throws Exception {
+        Assert.assertEquals("s3", manager.getEngine());
+    }
+
+    @BeforeClass
+    public static void startServer() throws Exception {
+        Thread.sleep(1000);
+        LOG = LoggerFactory.getLogger(S3DDFManagerTests.class);
+        S3DataSourceDescriptor  s3dsd = new S3DataSourceDescriptor(new S3DataSourceURI(""),
+            new S3DataSourceCredentials(System.getenv("AWS_ACCESS_KEY_ID"), System.getenv("AWS_SECRET_ACCESS_KEY")),
+            null,
+            null);
+        manager = (S3DDFManager)DDFManager.get(DDFManager.EngineType.S3, s3dsd);
+    }
+
+    @Test
+    public void testListing() throws DDFException {
+        List<String> buckets = manager.listBuckets();
+        System.out.println("========== buckets ==========");
+        for (String bucket: buckets) {
+            System.out.println(bucket);
+        }
+
+        System.out.println("========== jing-bucket/testFolder/ ==========");
+        List<String> keys = manager.listFiles("jing-bucket", "testFolder/");
+        for (String key: keys) {
+            System.out.println(key);
+        }
+
+        System.out.println("========== jing-bucket/testFolder/a.json ==========");
+        keys = manager.listFiles("jing-bucket", "testFolder/a.json");
+        for (String key: keys) {
+            System.out.println(key);
+        }
+    }
+
+    @Test
+    public void testCreateDDF() throws DDFException {
```

Comment:
Json files doesn't have header, only apply for tsv and csv
For the 2nd question, I don't know, but you can try, when a specify a non-existed path, it will create a null file and load from s3 still work, but return empty ddf